### PR TITLE
BAU sonarcloud report on PR

### PR DESCRIPTION
**Pull request by idrop**
## Proposed changes

Add a sonarqube report when PRs are raised. Check out the sonarcloud reports below.

### What changed

Update gradle, github actions.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -18,5 +19,20 @@ jobs:
...
+      - name: Cache Gradle packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (Wynndow) [inline .github/workflows/test.yml]: Where do these secrets live - I'm guessing they're set in the GitHub UI somewhere?
Could you add a comment describing what the GitHub token is called and which bot account owns it, just so it's easy to track down if it needs rotating at any point? Also perhaps something similar for the Sonar token?

Comment (idrop) [reply .github/workflows/test.yml]: I'll add to team manual if that's ok?

Comment (Wynndow) [reply .github/workflows/test.yml]: I'd be more keen to see it here, or at least a pointer to the team manual? If someone is looking at this in the future who isn't familiar with the page in the team manual they may not think to go looking there.

I'm just remembering the issue with tracking stuff down when the Travis incident happened.

Comment (idrop) [reply .github/workflows/test.yml]: @Wynndow I've removed the GITHUB_TOKEN with no discernable bad effect, and commented on the SONAR_TOKEN in the script

Comment (Wynndow) [reply .github/workflows/test.yml]: https://media3.giphy.com/media/d6nvFKK6Dr9YbBDeNQ/giphy.gif

**Review by Wynndow — APPROVED** [review]

**Pull request merged**